src/java.base/share/classes/java/lang/ModuleLayer.java

```diff
@@ -906,7 +909,7 @@ public static ModuleLayer empty() {
      *
      * @return The boot layer
      */
-    public static ModuleLayer boot() {
+    public static @Nullable ModuleLayer boot() {
```

Comment:
I'm a little tempted to leave this non-null, since the conditions under which it's `null` sound so rare. Maybe user code can't even observe them? But it's not like I expect [anyone](https://github.com/hibernate/hibernate-demos/blob/45d85e428b6231e55be172340d9a98e4c1c68b24/java9/custom-jlink-plugin/agent/src/main/java/org/hibernate/demos/jlink/agent/JLinkPluginRegistrationAgent.java#L27-L28) to call this method, anyway, so we might as well be conservative :)